src/auth/auth.controller.ts

```diff
@@ -75,4 +75,77 @@ export class AuthController {
       changePasswordRequest.newPassword,
     );
   }
+
+  @HttpCode(201)
+  @ApiOperation({
+    summary:
+      'Creates QR code for user to scan it for auth app (like google Authenticator)',
+  })
+  @UseGuards(JwtAuthGuard)
+  @Post('create-qr-2fa')
+  async createQrcodeFor2FA(
+    @UserId() userId: number,
+  ): Promise<CreateQrcodeFor2FA> {
+    const qrcode = await this.authService.createQrcodeFor2FA(userId);
+
+    return CreateQrcodeFor2FA.from(qrcode);
+  }
+
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Enables 2FA authentication for user',
+  })
+  @UseGuards(JwtAuthGuard)
+  @Post('enable-2fa')
+  async enable2FA(
+    @UserId() userId: number,
+    @Body() twoFactorAuthTokenData: Verify2FARequest,
+  ): Promise<Enable2FARespnse> {
+    const recoveryKeys = await this.authService.enable2FA(
+      userId,
+      twoFactorAuthTokenData.token,
+    );
+
+    return Enable2FARespnse.from(recoveryKeys);
+  }
+
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Disables 2FA for logged user' })
+  @UseGuards(JwtAuthGuard)
+  @Post('disable-2fa')
+  disable2FA(@UserId() userId: number): void {
+    this.authService.disable2FA(userId);
+  }
```

Comment:
I wonder if these two endpoints should be in `user` module instead of `auth` module? That is if we treat `auth` module as a module containing authentication / authorization methods. If you want to treat `auth module as module containing everything related to authentication / authorization (so including password reset, email change etc.) then I guess it's ok.

Reply:
I will treat this module as everything related to auth 😅